src/zc/intid/subscribers.py

```diff
@@ -0,0 +1,192 @@
+which, again, is a condition of the implementation using a
+``OIBTree.``
+
+A consequence of avoiding these references is that generally
+persistent objects that are expected to have intids assigned *should
+not* be used as keys in an ``OxBTree`` or stored in an ``OOSet.``
+Instead, all such data structures *should* use the integer
+variations (e.g., ``IISet``), with the intid as the key.
+
+As a corollary to the previous point, this module *must* be used
+with the intid utility from :mod:`zc.intid.utility`, (one
+implementing :class:`zc.intid.interfaces.IIntIds`), which does not
+depend on being able to use objects as keys in a BTree.
+
+Therefore, this module looks for utilities registered for that
+interface, not the :class:`zope.intid.interfaces.IIntIds`.
+
+We do, however, keep a few things in common:
+
+#. We do ensure that the object can be adapted to :class:`zope.keyreference.interface.IKeyReference`
+    In the common case of persistent objects, this will ensure that the
+    object is in the database and has a jar and oid, common needs.
+#. We do broadcast the events from :mod:`zope.intid.interfaces`, even though
+    the :mod:`zc.intid` package will broadcast its own events.
+    There seems to be no reason not to and things like zope.catalog
+    need them.
+
+Configuring
+===========
+
+To configure, you need to include ``subscribers.zcml``:
+
+.. code-block:: xml
+
+    <!-- configure.zcml -->
+    <!--
+    If we load zope.intid, we get subscribers for the Object events
+    that ensure all ILocation objects are registered/unregistered when
+    they are added/removed, plus another set of events when they
+    get/lose intids. This second set of events is meant to update
+    zope.catalog. A consequence of this is that ILocation objects must
+    be adaptable to KeyReferences when they are ObjectAdded (for
+    purposes of zope.intid, which we don't care about, but this also
+    ensures that they have ZODB Connections, which is good).
+
+    We cannot use these subscribers as-is due to the way the use IKeyReference
+    and try to register that. However, our subscribers *do* make sure that
+    the given objects can be adapted to IKeyReference because that's useful and
+    may be required by catalogs or other subscribers.
+    -->
+    <exclude package="zope.intid" file="subscribers.zcml" />
+    <include package="zope.intid" />
+
+    <include package="zope.keyreference" />
+
+    <!--
+    zc.intid fires a different set of events when objects gain/lose
+    intids.
+    -->
+    <include package="zc.intid" />
+
+    <!--
+    Make zc.intid utilities compatible with zope.intid utilities.
+    -->
+    <include package="zc.intid" file="zope-intid.zcml" />
+
+    <!-- To hook them up to the Object events, we need to include the file -->
+    <include package="zc.intid" file="subscribers.zcml" />
+
+"""
+from __future__ import print_function, absolute_import, division
```

Comment:
Thats just a part of the standard header my organization uses, which is where this code came from originally. Like the `#-*- coding: utf-8 -*-` line. It's just one way we make sure to write Python2/3 code. The imports are always there so that those features are always available.